src/ReportManager/SchemaFormContext/SchemaForm/fields/Text/index.js

```diff
@@ -0,0 +1,41 @@
+import React, { useContext } from 'react';
+
+import { SchemaFormContext } from '../../SchemaFormContext';
+import { isSchemaFieldRequired } from '../../SchemaFormContext/utils';
+import useFieldDetails from '../../useFieldDetails';
+
+const VALIDATION_ERROR_TYPES = {
+  REQUIRED: 'REQUIRED'
+};
+
+export const TextFieldValidators = {
+  [VALIDATION_ERROR_TYPES.REQUIRED]: (fieldValue, fieldName, schema) => {
+    return isSchemaFieldRequired(schema, fieldName) ? !fieldValue : false;
+  }
+};
+
+const Text = ({ fieldName }) => {
+  const { onFieldChange } = useContext(SchemaFormContext);
+  const textFieldDetails = useFieldDetails(fieldName);
+
+  const handleOnChange = (e) => onFieldChange(fieldName, e.currentTarget.value);
```

Comment:
nit: `e` is not a great variable name. I know it's common to call it like that but still reduces readability of the code.

Reply:
Done